### PR TITLE
Secret-storage no unlock fix

### PR DESCRIPTION
**Pull request by therandomstring**
A small bugfix concerning proper shutdown.
the bug concerns the mutex guarding the initialisation of secret-storage, where the mutex has been locked, but no unlock is present.

### CODE REVIEW

**Review by MrAnno — COMMENTED** [review]
What do you intend to fix with this commit?
Mutexes and atomic counters have different purposes, guarantees and costs.
Using an atomic counter here is plenty enough in my opinion, and serves its purpose.

File: lib/gprocess.c [hunk 1]
```diff
@@ -1543,6 +1544,7 @@ g_process_finish(void)
   */
 #endif
 
+  free(process_opts.argv_orig);
```
Comment (bazsi) [inline lib/gprocess.c:1547]: There's a comment right above this new line that explains why we don't want to free that line. I am more than happy to prove that this comment is already stale and we can actually perform this free operation, but please remove the comment along with it.

**Comment by therandomstring** [comment]
@MrAnno this fixes an issue while testing with libFuzzer, it hanging on the second run, as it tries to lock the mutex/counter not unlocked at the end of the previous test.
As per your request, I will revert back to an atomic counter.

File: lib/gprocess.c (hunk 1 above)
Comment (therandomstring) [reply lib/gprocess.c:1547]: Said comment is the reason, why I only freed this variable, and not `**environ`. However the fact that somehow I placed the free outside the if clause is proof enough that I should have drank at least two more cups of coffee yesterday.
I would be happy to investigate this leak further, but given the circumstances, I am not confident enough to still include this in this merge request, as the other fix is instrumental to the fuzzing project, so I do not want to delay that one.
Sorry for the blunder.

**Comment by kira-syslogng** [comment]
Build FAILURE

**Review by MrAnno — APPROVED** [review]

**Pull request merged**